addons/sale_timesheet/views/project_portal_templates.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="portal_timesheet_table_inherit" inherit_id="hr_timesheet.portal_timesheet_table">
+        <xpath expr="//thead/tr/th[@t-if='is_uom_day']" position="before">
+	    <t t-set="display_sol" t-value="False"/>
+	    <t t-foreach="timesheets" t-as="timesheet">
+	        <t t-if="timesheet.so_line != task.sale_line_id">
+		    <t t-set="display_sol" t-value="True"/>
+		</t>
+	    </t>
+            <th t-if="display_sol">Sales Order Item</th>
+        </xpath>
+        <xpath expr="//tr/td[t[@t-esc='timesheet.name']]" position="after">
+            <td  t-if="display_sol">
+	        <t t-if="timesheet.so_line.order_id.id"><a t-att-href="'/my/orders/%s' % timesheet.so_line.order_id.id"><t t-esc="timesheet.so_line.display_name"/></a></t>
```

Comment:
```suggestion
	        <t t-if="timesheet.so_line"><a t-att-href="'/my/orders/%s' % timesheet.so_line.order_id.id"><t t-out="timesheet.so_line.display_name"/></a></t>
```
Since `order_id` is a mandatory field on `sale.order.line`, I don't think it need to be tested.
Also, `t-esc`is deprecated for most usages, use `t-out` instead